### PR TITLE
net: Don't load the placeholder image for background images, only for image fragments.

### PR DESCRIPTION
**Pull request by pcwalton**
r? @jdm

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/5586)

### CODE REVIEW

**Comment by jdm** [comment]
```
failures:
    no-image.html == no-image-ref.html
```

**Comment by pcwalton** [comment]
I think I'll wait on #5767 to land first, because the test failure is a Linux-only race that's tightly coupled to the image cache task.

File: components/net_traits/image/holder.rs [hunk 1]
```diff
@@ -0,0 +1,104 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (glennw) [inline components/net_traits/image/holder.rs]: This whole file can be removed.

File: components/net_traits/local_image_cache.rs [hunk 2]
```diff
@@ -0,0 +1,174 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (glennw) [inline components/net_traits/local_image_cache.rs]: This whole file can be removed.

**Comment by pcwalton** [comment]
Review comments addressed.

@bors-servo: r=glennw

**Comment by jdm** [comment]
```
failures:
    no_image_background_a.html == no_image_background_ref.html
```

**Comment by jdm** [comment]
```
failures:
    no_image_background_a.html == no_image_background_ref.html
```

**Comment by pcwalton** [comment]
@bors-servo: r=glennw

Apologies for trying again, but I've run the tests multiple times on both Mac and Linux and have been unable to reproduce. I want to rule out the possibility that this was a manifestation of the flakiness that @glennw fixed. If this fails again I will SSH into the bots to try to reproduce.

**Pull request merged**